ddio/lnxkey_sdl.cpp

```diff
@@ -339,38 +339,38 @@ static inline uint8_t sdlkeycode_to_keycode(uint32_t sdlkeycode) {
   return (uint8_t)rc;
 }
 
-int sdlKeyFilter(const SDL_Event *event) {
+bool sdlKeyFilter(const SDL_Event *event) {
   uint8_t kc = 0;
 
-  if ((event->type != SDL_KEYUP) && (event->type != SDL_KEYDOWN))
-    return (1);
+  if ((event->type != SDL_EVENT_KEY_UP) && (event->type != SDL_EVENT_KEY_DOWN))
+    return true;
 
-  switch (event->key.state) {
-  case SDL_PRESSED:
+  switch (event->key.down) {
```

Comment:
Since `event->key.down` is a bool now, the `switch` seems odd and might be changed to an `if`.

Reply:
reworked to use if/else logic